packages/cinterop/src/nativeDarwin/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -516,21 +515,75 @@ actual object RealmInterop {
         //      val dispatcher = runBlocking { coroutineContext[CoroutineDispatcher.Key] }
         //  but requires opting in for @ExperimentalStdlibApi, and have really gotten it to play
         //  for default cases.
-        if (dispatcher != null) {
-            val scheduler = checkedPointerResult(createSingleThreadDispatcherScheduler(dispatcher))
-            realm_wrapper.realm_config_set_scheduler(config.cptr(), scheduler)
-        } else {
-            // If there is no notification dispatcher use the default scheduler.
-            // Re-verify if this is actually needed when notification scheduler is fully in place.
-            val scheduler = checkedPointerResult(realm_wrapper.realm_scheduler_make_default())
-            realm_wrapper.realm_config_set_scheduler(config.cptr(), scheduler)
-        }
+        realm_wrapper.realm_config_set_scheduler(config.cptr(), scheduler.cptr())
+
         val realmPtr = CPointerWrapper<LiveRealmT>(realm_wrapper.realm_open(config.cptr()))
         // Ensure that we can read version information, etc.
         realm_begin_read(realmPtr)
         return Pair(realmPtr, fileCreated.value)
     }
 
+    actual fun realm_create_scheduler(): RealmSchedulerPointer {
+        // If there is no notification dispatcher use the default scheduler.
+        // Re-verify if this is actually needed when notification scheduler is fully in place.
+        val scheduler = checkedPointerResult(realm_wrapper.realm_scheduler_make_default())
+        return CPointerWrapper<RealmSchedulerT>(scheduler)
+    }
+
+    actual fun realm_create_scheduler(dispatcher: CoroutineDispatcher): RealmSchedulerPointer {
+        printlntid("createSingleThreadDispatcherScheduler")
```

Comment:
Debug output that should be removed? Same for a lot of other places

Reply:
This is it a no-op lambda unless redefined. It was left this way as an easy way to reenable debugging, as it is quite tricky to otherwise get an overview of all involved scheduling-related operations.